compiler/rustc_ast/src/visit.rs

```diff
@@ -32,6 +32,13 @@ pub enum FnCtxt {
     Assoc(AssocCtxt),
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum BoundCtxt {
+    Normal,
```

Comment:
Can you leave some comments for what these mean in AST/HIR? Like "normal" is `<T: Bound>` and `where T: Bound` or something?